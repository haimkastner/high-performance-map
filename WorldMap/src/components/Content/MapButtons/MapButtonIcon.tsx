import React from "react";
import { Box } from "@material-ui/core";

/**
 * the properties for an Icon in one of the map buttons. 
 */
interface Props {
  imageName: string;
  width?: string;
  height?: string;
}

/**
 * Gets the @param props to put on the button's image.
 * @returns the image with the updated css properties.
 */
export class MapButtonIcon extends React.Component<any, Props> {
  render() {
    return (
      <Box>
        <img
          width={this.props.width ?? 25} //if not given, default is 25
          height={this.props.height ?? 25} //if not given, default is 25
          src={require(`./../../../assets/img/${this.props.imageName}`)} //the complete path with the name
        />
      </Box>
    );
  }
}
