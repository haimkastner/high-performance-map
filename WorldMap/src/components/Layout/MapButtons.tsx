import React from 'react';
import styles from './Styles/MapButtons.module.css';

export class MapButtons extends React.Component {

    render() {
        return (<h1 className={styles.Downbar}>MapButtons</h1>);
    }
}
