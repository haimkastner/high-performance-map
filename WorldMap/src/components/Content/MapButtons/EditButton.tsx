import React from "react";
import { MapButtonIcon } from "./MapButtonIcon";
import { Box, Button } from "@material-ui/core";
import { setSharedState, subsribeState, unsubsribeState } from "../../../services/stats-store";

interface State {
  speed: string;
}

export class EditButton extends React.Component<any, State> {
  token: any;

  state: State = {
    speed: "111"
  };

  publish() {
    setSharedState("speed", 99);
    setSharedState("direction", 100);
  }

  componentDidMount() {
    this.token = subsribeState("speed", this);
  }
  componentWillUnmount() {
    unsubsribeState(this.token);
  }

  render() {
    return (
      <Box>
        <Button onClick={this.publish}>
          {this.state.speed}
          <MapButtonIcon
            imageName={"display-copy-shape-2@2x.png"}
          ></MapButtonIcon>
        </Button>
      </Box>
    );
  }
}
