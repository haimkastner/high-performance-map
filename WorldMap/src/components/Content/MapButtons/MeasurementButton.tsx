import React from "react";
import { MapButtonIcon } from './MapButtonIcon';

export class MeasurementButton extends React.Component {
  render() {
    return (<MapButtonIcon imageName={"adding-copy-shape-3@2x.png"}></MapButtonIcon>);
  }
}
