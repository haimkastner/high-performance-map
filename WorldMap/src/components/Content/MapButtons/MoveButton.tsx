import React from "react";
import { MapButtonIcon } from "./MapButtonIcon";
import { Box, Button } from "@material-ui/core";
import { setSharedState } from "../../../services/stats-store";
import { Platform } from "../../../models/models";

export class MoveButton extends React.Component {
  dir = 50;
  selectPlatform = () => {
    const platform: Platform = {
      ID: 5555,
      Name: "5555",
      PlatformType: "Aircraft",
      Spacial: {
        Movement: {
          Course: this.dir++,
          Speed: 400
        },
        Position: {
          Altitude: 0,
          Latitude: 70,
          Longitude: 44
        }
      }
    };
    setSharedState("selectedPlatform", platform);
    setSharedState("showNavBar", true);
    
  };

  render() {
    return (
      <Box>
        <Button onClick={this.selectPlatform}>
          <MapButtonIcon
            imageName={"sensors-move@2x.png"}
            width="19"
            height="19"
          ></MapButtonIcon>
        </Button>
      </Box>
    );
  }
}
