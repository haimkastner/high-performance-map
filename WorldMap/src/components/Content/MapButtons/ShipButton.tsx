import React from 'react';
import { MapButtonIcon } from './MapButtonIcon';

export class ShipButton extends React.Component {
  public render() {
    return <MapButtonIcon imageName={'movement-path-4333-shape-1@2x.png'}></MapButtonIcon>;
  }
}
