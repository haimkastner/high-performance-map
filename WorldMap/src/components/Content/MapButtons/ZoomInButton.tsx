import React from 'react';
import { MapButtonIcon } from './MapButtonIcon';

export class ZoomInButton extends React.Component {
  public render() {
    return <MapButtonIcon imageName={'side-menu-shape-copy-2@2x.png'}></MapButtonIcon>;
  }
}
