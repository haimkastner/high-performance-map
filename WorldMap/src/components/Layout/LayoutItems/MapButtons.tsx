import { createStyles, Grid, makeStyles, Theme } from '@material-ui/core';
import React from 'react';
import { EditButton } from '../../Content/MapButtons/EditButton';
import { MeasurementButton } from '../../Content/MapButtons/MeasurementButton';
import { MoveButton } from '../../Content/MapButtons/MoveButton';
import { ShipButton } from '../../Content/MapButtons/ShipButton';
import { ZoomInButton } from '../../Content/MapButtons/ZoomInButton';
import { ZoomOutButton } from '../../Content/MapButtons/ZoomOutButton';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      height: '100%',
      width: '100%',
    },
  }),
);

export function MapButtons() {
  const classes = useStyles();

  return (
    <Grid className={classes.root} container direction="column" justify="space-around" alignItems="center">
      <MoveButton />
      <MeasurementButton />
      <EditButton />
      <ShipButton />
      <ZoomInButton />
      <ZoomOutButton />
    </Grid>
  );
}
