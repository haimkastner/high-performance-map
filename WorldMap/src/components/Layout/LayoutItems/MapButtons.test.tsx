import React from "react";
import ReactDOM from "react-dom";
import { MapButtons } from "./MapButtons";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("MapButtons Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<MapButtons />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("matches the snapshot", () => {
    const tree = renderer.create(<MapButtons />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("MapButtons Component Logic", () => {

  });
});
