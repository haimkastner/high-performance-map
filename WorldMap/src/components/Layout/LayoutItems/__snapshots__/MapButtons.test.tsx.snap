// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapButtons Component matches the snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-root-146 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
>
  <div
    className="MuiBox-root MuiBox-root-251"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <div
          className="MuiBox-root MuiBox-root-284"
        >
          <img
            height="19"
            src="sensors-move@2x.png"
            width="19"
          />
        </div>
      </span>
    </button>
  </div>
  <div
    className="MuiBox-root MuiBox-root-285"
  >
    <img
      height={25}
      src="adding-copy-shape-3@2x.png"
      width={25}
    />
  </div>
  <div
    className="MuiBox-root MuiBox-root-286"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <div
          className="MuiBox-root MuiBox-root-287"
        >
          <img
            height={25}
            src="display-copy-shape-2@2x.png"
            width={25}
          />
        </div>
      </span>
    </button>
  </div>
  <div
    className="MuiBox-root MuiBox-root-288"
  >
    <img
      height={25}
      src="movement-path-4333-shape-1@2x.png"
      width={25}
    />
  </div>
  <div
    className="MuiBox-root MuiBox-root-289"
  >
    <img
      height={25}
      src="side-menu-shape-copy-2@2x.png"
      width={25}
    />
  </div>
  <div
    className="MuiBox-root MuiBox-root-290"
  >
    <img
      height={25}
      src="side-menu-shape-copy-2@2x.png"
      width={25}
    />
  </div>
</div>
`;
